Sources/NIO/Array+UInt8+ByteBuffer.swift

```diff
@@ -0,0 +1,23 @@
+//===----------------------------------------------------------------------===//
```

Comment:
can we put the `String` and the `Array` stuff in a file `ByteBuffer-conversions.swift` or so? Otherwise they're in totally weird places in the project and not next to the other ByteBuffer ones. Also, fewer files is better if they're that small.